ls1intum/Artemis
# `Development`: Fast Angular 16 build

based on #6544

### Checklist
#### General



#### Client

### Motivation and Context


[Angular 16 is here](https://blog.angular.io/angular-v16-is-here-4d7a28ec680d)! With it it [brings a new Vite development server](https://angular.io/guide/esbuild#vite-as-a-development-server). This development server is much faster than the current one - which will drastically increase developer productivity! 🚀

### Description

First of all, this PR updates Angular to version 16.

To be able to leverage the new Angular Build capabilities, I had to remove the current Webpack configuration because well, it's specific to Webpack. Instead, now all configuration should happen in `angular.json`. This was not possible for
- dynamic environment variables and
- merging the i18n json files

Because of this, I introduced a `prebuild.mjs` script.

### Steps for Testing

- pull from the branch
- `npm install` (you might have to delete node_modules first if it errors)
- start the server and the client
- try the live reload functionality
- click around in the web app to see whether anything is broken

### Review Progress






#### Code Review
#### Manual Tests

### Screenshots



Screen recording of me removing and re-adding the refresh button on the course page. Previously, my build times were around 16x as long (4x with some build speed improvement tricks):


https://user-images.githubusercontent.com/9006596/236708780-76c7598b-678c-4b18-ad31-7d7af44181d5.mp4
